### DISCUSSION TITLE
Build multiple

### DISCUSSION THREAD

**Discussion by sosie-js** [keplerproject/luarocks#1698] [Ideas]
Hello here, is it possible at the same time to build a module and a command?
I mean using luke
```
build = {
   type = "command",
   build_command  = "./build-aux/luke lukefile install all",
   copy_directories = {
      "doc"
   }
}
```
and
```
build = {
   type = "builtin",
   modules = {
      main = "src/main.lua"
   }
}
```

in a single rockspec?

Thanks for your response.

**Comment by sosie-js** [comment]
I see there is quite a similar discussion here about naming conventions with differents archs https://github.com/luarocks/luarocks/issues/786#issuecomment-475210467 but it not quite the same as here we are talking about luarock build types

**Comment by hishamhm** [comment]
@sosie-js no, it's not possible. if you want to perform both installations from a single command, you'd have to create two rockspecs, and have one be a dependency of the other. (But yes, this might end up causing the same source repo to be downloaded twice when you run `luarocks build`...)

Alternatively, you could create a custom build type which runs two build engines, name it something like `luarocks-build-mybuildtype` and then load that as a `build_dependency`. But I think that's too much trouble.

**Reply by sosie-js** [reply]
So that means i I declare the rockspec with the same name in to different directories and the one in a subdirectory is a dependency of other and has a higher version (even) number, it is that?

**Reply by hishamhm** [reply]
No, the rockspecs would need to have different names, effectively producing two distinct packages with a dependency relation.

**Reply by sosie-js** [reply]
@hishamhm , two good news for you

1)  I solved it simply like:
```lua
build = {
   type = "builtin",
   modules = {
      peagisub = "src/peagisub.lua" <--- the builtin
   },
   install= {
	bin = {
	  ["peagisub"] = "src/peagisub" <-- the command to call the builtin
	} 
    },
    copy_directories = { "config" }
}

```
2) I am about to release a helper around this.

Your stuff rocks!

**Reply by hishamhm** [reply]
ah, yes, if all you need is to install `.lua` files, then the `builtin` mode works just fine for Lua modules + bin scripts!

**Reply by sosie-js** [reply]
And behind all of this is the wrap_script jewel. When grabbed we can build a nice custom command with luarock builtins as backend. Let's take the example of peagisub 1.5 thats bring the jewel in front with shell  or aegisub frontends

![peagisub overview](https://github.com/sosie-js/peagisub-vs/blob/1.4.0/doc/Overview.png?raw=true#)
